File: android/src/main/java/com/lightningkite/khrysalistemplate/layouts/ExternalTestXml.kt
//
// ExternalTestXml.kt
// Created by Khrysalis XML Android
//
package com.lightningkite.khrysalistemplate.layouts

import android.widget.*
import android.view.*
import com.lightningkite.khrysalis.views.*
import com.lightningkite.khrysalistemplate.R

class ExternalTestXml {

    lateinit var scrollView: ScrollView
    lateinit var openMap: Button
    lateinit var openWeb: Button
    lateinit var openEvent: Button
    
    
    lateinit var xmlRoot: View

    fun setup(dependency: ViewDependency): View {
        val view = LayoutInflater.from(dependency.context).inflate(R.layout.external_test, null, false)
        return setup(view)
    }
    fun setup(view: View): View {
        xmlRoot = view
        scrollView = view.findViewById<ScrollView>(R.id.scrollView)
        openMap = view.findViewById<Button>(R.id.openMap)
        openWeb = view.findViewById<Button>(R.id.openWeb)
        openEvent = view.findViewById<Button>(R.id.openEvent)
        
        
        
        return view
    }
}

File: web/src/layout/ExternalTestXml.ts
//
// ExternalTestXml.ts
// Created by Khrysalis XML Typescript
//
import { loadHtmlFromString, findViewById, getViewById, replaceViewWithId, startupAutoResize } from 'butterfly-web/dist/views/html'
import { customViewSetDelegate } from 'butterfly-web/dist/views/CustomView'
import htmlForDefault from './external_test.html'
//! Declares com.lightningkite.butterflytemplate.layouts.ExternalTestXml
export class ExternalTestXml {
    xmlRoot!: HTMLElement
    scrollView!: HTMLDivElement
    openMap!: HTMLButtonElement
    openWeb!: HTMLButtonElement
    openEvent!: HTMLButtonElement
    loadHtmlString(): string {
        return htmlForDefault
    }
    setup(dependency: Window): HTMLElement {
        const view = loadHtmlFromString(this.loadHtmlString())
        this.xmlRoot = view
        this.scrollView = getViewById<HTMLDivElement>(view, "scrollView")
        this.openMap = getViewById<HTMLButtonElement>(view, "openMap")
        this.openWeb = getViewById<HTMLButtonElement>(view, "openWeb")
        this.openEvent = getViewById<HTMLButtonElement>(view, "openEvent")
        return view
    }
}
